src/test/java/teammates/test/cases/webapi/CreateFeedbackResponseActionTest.java

```diff
@@ -23,13 +50,167 @@ protected String getRequestMethod() {
+        assertNotNull(typicalDataInstructor.getFeedbackResponseId());
+        assertEquals(instructor2OfCourse1.getEmail(), typicalDataInstructor.getGiverIdentifier());
+        assertEquals(instructor2OfCourse1.getEmail(), typicalDataInstructor.getRecipientIdentifier());
+
+        ______TS("response already exists");
+        //show that this FeedbackResponse already exists
+        FeedbackResponseAttributes existingFeedbackResponse =
+                logic.getFeedbackResponse(qn2InSession1InCourse1.getId(), student2InCourse1.getEmail(),
+                        student5InCourse1.getEmail());
+        assertNotNull(existingFeedbackResponse);
+
+        String[] paramsQ2S1C1 = {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, qn2InSession1InCourse1.getId(),
+                Const.ParamsNames.INTENT, Intent.STUDENT_SUBMISSION.toString(),
+        };
+        loginAsStudent(student2InCourse1.getGoogleId());
+        FeedbackResponseCreateRequest createRequestAlreadyExists = getResponseRequest(student5InCourse1.getEmail());
+        CreateFeedbackResponseAction typicalActionAlreadyExists = getAction(createRequestAlreadyExists, paramsQ2S1C1);
+        assertThrows(InvalidHttpRequestBodyException.class, () -> getJsonResult(typicalActionAlreadyExists));
+
+        ______TS("invalid intent");
+        String[] invalidIntentParams = {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, qn1InSession1InCourse1.getId(),
+                Const.ParamsNames.INTENT, Intent.FULL_DETAIL.toString(),
+        };
+        verifyHttpParameterFailure(invalidIntentParams);
     }
 
     @Test
     @Override
     protected void testAccessControl() throws Exception {
-        // TODO
+        ______TS("non-exist feedback question");
+        loginAsInstructor(instructor1OfCourse1.getGoogleId());
+        String[] nonExistFeedbackQuestionParams = {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, "randomNonExist",
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+        };
+        assertThrows(EntityNotFoundException.class, () -> getAction(nonExistFeedbackQuestionParams)
+                .checkAccessControl());
+        ______TS("feedback session is closed");
+        String[] closedFeedbackSessionParams = {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, qn1InClosedSessionInCourse1.getId(),
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+        };
+        verifyCannotAccess(closedFeedbackSessionParams);
+        ______TS("in preview request");
+        String[] previewParams = {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, qn1InSession1InCourse1.getId(),
+                Const.ParamsNames.INTENT, Intent.STUDENT_SUBMISSION.toString(),
+                Const.ParamsNames.PREVIEWAS, "nonEmptyPreviewParam",
+        };
+        verifyCannotAccess(previewParams);
+        ______TS("not answerable for students");
```

Comment:
I think it is better to have a verification to check that the question is not answerable for students.